opensearch-operator/pkg/builders/cluster.go

```diff
@@ -157,6 +162,57 @@ func NewSTSForNodePool(
 		ProbeHandler:        corev1.ProbeHandler{TCPSocket: &corev1.TCPSocketAction{Port: intstr.IntOrString{IntVal: cr.Spec.General.HttpPort}}},
 	}
 
+	envVars := []corev1.EnvVar{
+		{
+			Name:  "cluster.initial_master_nodes",
+			Value: BootstrapPodName(cr),
+		},
+		{
+			Name:  "discovery.seed_hosts",
+			Value: DiscoveryServiceName(cr),
+		},
+		{
+			Name:  "cluster.name",
+			Value: cr.Name,
+		},
+		{
+			Name:  "network.bind_host",
+			Value: "0.0.0.0",
+		},
+		{
+			// Make elasticsearch announce its hostname instead of IP so that certificates using the hostname can be verified
+			Name:      "network.publish_host",
+			ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "metadata.name"}},
+		},
+		{
+			Name:  "OPENSEARCH_JAVA_OPTS",
+			Value: jvm,
+		},
+		{
+			Name:  "node.roles",
+			Value: strings.Join(selectedRoles, ","),
+		},
+		{
+			Name:  "OPENSEARCH_USER",
+			Value: username,
+		},
+		{
+			Name: "OPENSEARCH_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: fmt.Sprintf("%s-admin-password", cr.Name),
+					},
+					Key: "password",
+				},
+			},
+		},
+	}
+	// cr.Spec.NodePool.envVars
```

Comment:
Hey :) 
this is the only thing that you added in that file ?

Reply:
Hey. 

I turned the inline `EnvVar[]` into variable so that additional / user-defined values could be appended to it. 

If that's what you mean?

Reply:
I think I see what you're getting at. I put the inline initialization back, and just appended it `StatefulSet` variable (like you're already doing w/ `extraConfig`.